desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -15,11 +15,72 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+import json
+from pprint import pprint
+
 from django.shortcuts import render
 from django.http import HttpResponse
-from desktop.lib.django_util import login_notrequired
+from desktop.lib.django_util import login_notrequired, JsonResponse
+from desktop import conf
+from django.views.decorators.csrf import csrf_exempt
+
+from django.conf import settings
+from slack_sdk import WebClient
+
+LOG = logging.getLogger(__name__)
+
+SLACK_VERIFICATION_TOKEN = conf.SLACK.SLACK_VERIFICATION_TOKEN.get()
+SLACK_BOT_USER_TOKEN = conf.SLACK.SLACK_BOT_USER_TOKEN.get()
+
+
+slack_client = WebClient(token=SLACK_BOT_USER_TOKEN)
+appname = "hue_bot"
 
 
 @login_notrequired
-def home(request):
-  return HttpResponse("Hello World")
\ No newline at end of file
+@csrf_exempt
+def slack_events(request):
+  slack_message = json.loads(request.body.decode('utf-8'))
+
+  if slack_message['token'] != SLACK_VERIFICATION_TOKEN:
+    return HttpResponse(status=403)
+
+  # challenge verification
+  if slack_message['type'] == 'url_verification':
+    response_dict = {"challenge": slack_message.get('challenge')}
+    return JsonResponse(response_dict, status=200)
+  
+  if 'event' in slack_message:
+    event_message = slack_message['event']
+    parse_events(event_message)
+
+  return HttpResponse(status=200)
+
+def parse_events(event_message):
+  user_id = event_message.get('user')
+  text = event_message.get('text')
+  channel = event_message.get('channel')
+  BOT_ID = get_bot_id(appname)
+
+  # ignore bot's own message
+  if BOT_ID == user_id:
+    return HttpResponse(status=200)
+
+  if 'hello hue' in text.lower():
+    say_hi_user(channel, user_id)
+  
+def say_hi_user(channel, user_id):
+  """Bot sends Hi<username> message in a specific channel"""
+  bot_message = f'Hi <@{user_id}> :wave:'
+  response = slack_client.api_call(api_method='chat.postMessage', json={'channel': channel, 'text': bot_message})
+  if response["ok"]:
+    return HttpResponse(status=200)
+
+def get_bot_id(botusername):
+  """Takes in bot username, Returns the bot id"""
+  response = slack_client.api_call('users.list')
+  users = response['members']
+  for user in users:
+    if 'name' in user and botusername in user.get('name') and not user.get('deleted'):
```

Comment:
nit:

'name' in user and botusername in user.get('name')

-->

botusername in user.get('name', '')

Reply:
Fixed

Reply:
Done